test/core/utils/get-frame-context.js

```diff
@@ -0,0 +1,182 @@
+describe('utils.getFrameContexts', function () {
+  var getFrameContexts = axe.utils.getFrameContexts;
+  var fixture = document.querySelector('#fixture')
+
+  it('returns an empty array if the page has no frames', function () {
+    var frameContext = getFrameContexts()
+    assert.isArray(frameContext)
+    assert.lengthOf(frameContext, 0)
+  });
+
+  it('returns a `frameSelector` for each included frame', function () {
+    fixture.innerHTML =
+      '<iframe></iframe>' +
+      '<iframe></iframe>' +
+      '<iframe></iframe>'
+
+    var selectors = getFrameContexts().map(function (frameData) {
+      return frameData.frameSelector;
+    });
+    assert.include(selectors[0], 'iframe:nth-child(1)')
+    assert.include(selectors[1], 'iframe:nth-child(2)')
+    assert.include(selectors[2], 'iframe:nth-child(3)')
+    assert.lengthOf(selectors, 3)
+  });
+
+  it('returns a `frameContext` for each included frame', function () {
+    fixture.innerHTML =
+      '<iframe id="f1"></iframe>' +
+      '<iframe id="f2"></iframe>' +
+      '<iframe id="f3"></iframe>'
+    var context = {
+      include: [
+        ["#f1", "header"],
+        ["#f2", "main"]
+      ],
+      exclude: [
+        ["#f3", "footer"]
+      ]
+    }
+    var contexts = getFrameContexts(context).map(function (frameData) {
+      return frameData.frameContext;
+    });
+
+    assert.deepEqual(contexts[0], { include: [['header']], exclude: [] })
+    assert.deepEqual(contexts[1], { include: [['main']], exclude: [] })
+    assert.deepEqual(contexts[2], { include: [], exclude: [['footer']] })
+  });
+
+  it('returns a `frameContext` for each included frame', function () {
+    fixture.innerHTML =
+      '<iframe id="f1"></iframe>' +
+      '<iframe id="f2"></iframe>' +
+      '<iframe id="f3"></iframe>'
+    var context = {
+      include: [
+        ["#f1", "header"],
+        ["#f2", "main"]
+      ],
+      exclude: [
+        ["#f3", "footer"]
+      ]
+    }
+    var contexts = getFrameContexts(context).map(function (frameData) {
+      return frameData.frameContext;
+    });
+
+    assert.deepEqual(contexts[0], {
+      include: [['header']],
+      exclude: []
+    })
+    assert.deepEqual(contexts[1], {
+      include: [['main']],
+      exclude: []
+    })
+    assert.deepEqual(contexts[2], {
+      include: [],
+      exclude: [['footer']]
+    })
+  });
+  
+  it.skip('mixes contexts if the frame is selected twice', function () {
```

Comment:
There's a bug the presents this from passing. Need to fix that, but maybe in a separate PR.